pkg/registry/core/pod/rest/log.go

```diff
@@ -90,6 +91,11 @@ func (r *LogREST) Get(ctx context.Context, name string, opts runtime.Object) (ru
 
 	if !utilfeature.DefaultFeatureGate.Enabled(features.PodLogsQuerySplitStreams) {
 		logOpts.Stream = nil
+	} else {
```

Comment:
This is not right.  If the gate is enabled, defaults should have been set. 

pkg/apis/core/v1/defaults.go:
```
442 func SetDefaults_PodLogOptions(obj *v1.PodLogOptions) {                                                                          
443     if utilfeature.DefaultFeatureGate.Enabled(features.PodLogsQuerySplitStreams) {
444         if obj.Stream == nil {
445             obj.Stream = ptr.To(v1.LogStreamAll)
446         }
447     }
448 }
```

and

pkg/apis/core/v1/zz_generated.defaults.go:
```
51     scheme.AddTypeDefaultingFunc(&corev1.PodLogOptions{}, func(obj interface{}) { SetObjectDefaults_PodLogOptions(obj.(*corev1.P     odLogOptions)) })
```

So why is it not being called?

Reply:
> normal defaulting to params objects doesn't get applied if there are no query params

https://github.com/kubernetes/kubernetes/pull/128712#issuecomment-2465809232